File: 0080 Remove Duplicates from Sorted Array II/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() < 3) return nums.size();
        int k = 1;
        for (int i = 2; i < nums.size(); i ++ )
            if (nums[i] != nums[k - 1])
                nums[ ++ k] = nums[i];
        k ++ ;
        return k;
    }
};

int main(){
    Solution s;
    vector<int> vec{1,1,1,2,2,3};
    assert(s.removeDuplicates(vec) == 5);
    return 0;
}

File: 0080 Remove Duplicates from Sorted Array II/solution.py
#! python3

class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i = 0
        for n in nums:
            if i < 2 or n > nums[i - 2]:
                nums[i] = n
                i += 1
        return i

if __name__ == "__main__":
    print(Solution().removeDuplicates([1,1,1,2,2,3]))